client/src/components/NavigationBlocker/NavigationBlocker.js

```diff
@@ -0,0 +1,28 @@
+import { useEffect } from 'react';
+import PropTypes from 'prop-types';
+import { unstable_useBlocker as useBlocker } from 'react-router-dom';
+
+export default function NavigationBlocker({ shouldBlockFn, blockMessage }) {
+  const blocker = useBlocker(shouldBlockFn);
+  useEffect(() => {
```

Comment:
Has to be in an effect to avoid the console error I mentioned in https://github.com/silverstripe/silverstripe-admin/pull/1458#issuecomment-1439421893